projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -33,41 +43,66 @@ const getMediaDetails = media => {
 		mediaWidth: mediaObject.width,
 		mediaHeight: mediaObject.height,
 		mediaSourceUrl: mediaObject.source_url,
+		fullFileSize: media.media_details.filesize,
+		mime: media.mime_type,
 	};
 };
 
 const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { enabledConnections } = useSocialMediaConnections();
+
+	const { maxImageSize, getValidationError } = useMediaRestrictions( enabledConnections );
+	const allowedMediaTypes = getAllowedMediaTypes();
 
-	const mediaObject = useSelect( select =>
-		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
+	const mediaObject = useSelect(
+		select => select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } ),
+		[ attachedMedia[ 0 ] ]
 	);
+	const { mediaWidth, mediaHeight, mediaSourceUrl, fullFileSize, mime } = getMediaDetails(
+		mediaObject
+	);
+
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if ( ! mediaObject ) {
+			return;
+		}
 
-	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
+		const error = getValidationError( fullFileSize, mime );
+		if ( error ) {
+			setValidationError( error );
+			updateAttachedMedia( [] );
+		}
+	}, [ updateAttachedMedia, mediaObject, getValidationError, fullFileSize, mime ] );
```

Comment:
I'm just I'm just missing it, but what does this `useEffect` call do that isn't already handled by the validation check in `onUpdateMedia`? Is it for when media gets updated programmatically (e.g. with a script?)

Reply:
@danielpost I put in the `useEffect` to handle the scenario, when users enable/disable some connections. Without this, if they turn on a connection which has a smaller size allowed, it wouldn't remove the image not show the error. That said, it feels like this could be done differently without the `useEffect`, however I could not find a way that worked. Do you have an idea for this?

Reply:
That makes sense! The dependencies (`updateAttachedMedia, mediaObject, getValidationError, fullFileSize, mime`) don't really reflect this. Does a change in `enabledConnections` make a new `getValidationError` which in turn triggers this `useEffect`?

(I'm also wondering if you need `fullFileSize` and `mime` in there. Do those ever change without a change in `mediaObject`?)